src/Database/Mysql/Driver/Mysql.php

```diff
@@ -43,11 +51,17 @@ public function create(Tenant $tenant): bool
 
         event(new Events\Creating($tenant, $config, $this));
 
-        return $this->processAndDispatch(Events\Created::class, $tenant, [
-            'user'     => "CREATE USER IF NOT EXISTS `{$config['username']}`@'{$config['host']}' IDENTIFIED BY '{$config['password']}'",
-            'database' => "CREATE DATABASE `{$config['database']}`",
-            'grant'    => "GRANT ALL ON `{$config['database']}`.* TO `{$config['username']}`@'{$config['host']}'",
-        ]);
+        $result = $this->queryManager->setConnection($this->system($tenant))
+            ->processTransaction(function () use ($config) {
+                $this->statement("CREATE USER IF NOT EXISTS `{$config['username']}`@'{$config['host']}' IDENTIFIED BY '{$config['password']}'");
+                $this->statement("CREATE DATABASE `{$config['database']}`");
+                $this->statement("GRANT ALL ON `{$config['database']}`.* TO `{$config['username']}`@'{$config['host']}'");
```

Comment:
You can't really be sure that the statement function can be called in the closure? Interface does not define that function

Reply:
Crap, you're right... this is partly why I didn't want to go down the interface route.  Interfaces can't have private/protected methods and there's absolutely no reason for `statement` to be part of the public interface

Reply:
Will think about it, let's discuss this one on discord whenever we have ideas on how to fix this